### PR TITLE
Early basic implementation of the weapons list

### PR DESCRIPTION
**Pull request by ahctang**

### CODE REVIEW

File: app/src/main/java/com/gatheringhallstudios/mhworlddatabase/adapters/WeaponTypeAdapterDelegate.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import android.view.ViewGroup
+import com.gatheringhallstudios.mhworlddatabase.R
+import com.gatheringhallstudios.mhworlddatabase.adapters.common.SimpleListDelegate
+import com.gatheringhallstudios.mhworlddatabase.adapters.common.SimpleViewHolder
+import com.gatheringhallstudios.mhworlddatabase.assets.AssetLoader
+import com.gatheringhallstudios.mhworlddatabase.data.models.WeaponType
+import kotlinx.android.synthetic.main.listitem_large.*
+
+/**
+ * Defines an adapter delegate for a list of item combinations
```
Comment (jaysondc) [inline app/src/main/java/com/gatheringhallstudios/mhworlddatabase/adapters/WeaponTypeAdapterDelegate.kt]: Outdated comment.

File: app/src/main/java/com/gatheringhallstudios/mhworlddatabase/data/dao/WeaponDao.java [hunk 2]
```diff
@@ -1,21 +0,0 @@
...
-
-import android.arch.lifecycle.LiveData;
-import android.arch.persistence.room.Dao;
-import android.arch.persistence.room.Query;
-
-import com.gatheringhallstudios.mhworlddatabase.data.models.WeaponBasic;
-import com.gatheringhallstudios.mhworlddatabase.data.types.WeaponType;
-
-/**
- * Created by Carlos on 3/21/2018.
```
Comment (jaysondc) [inline app/src/main/java/com/gatheringhallstudios/mhworlddatabase/data/dao/WeaponDao.java]: Might as well update this one too. Put your name on it!

File: app/src/main/java/com/gatheringhallstudios/mhworlddatabase/data/dao/WeaponDao.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.gatheringhallstudios.mhworlddatabase.data.models.WeaponBase
+
+/**
+ * Created by Carlos on 3/21/2018.
+ */
+@Dao
+abstract class WeaponDao {
+
+//    @Query("SELECT DISTINCT w.weapon_type " +
```
Comment (jaysondc) [inline app/src/main/java/com/gatheringhallstudios/mhworlddatabase/data/dao/WeaponDao.kt]: Wait what's going on here? Are we using the kotlin or java version?

File: app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/weapons/list/WeaponListFragment.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                WeaponTypeModel(getString(R.string.heavy_bowgun_title), WeaponTypeEnum.HEAVY_BOWGUN),
+                WeaponTypeModel(getString(R.string.bow_title), WeaponTypeEnum.HEAVY_BOWGUN))
+
+        adapter.items = weaponTypes
+        adapter.notifyDataSetChanged()
+    }
+
+    class ViewModel(application: Application) : AndroidViewModel(application) {
+        private val dao = MHWDatabase.getDatabase(application).weaponDao()
+//        val weaponTypes = dao.loadWeaponTypes(AppSettings.dataLocale)
```
Comment (jaysondc) [inline app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/weapons/list/WeaponListFragment.kt]: Add a to-do if this is meant to be used for translations later.

**Comment by ahctang** [comment]
Changes made to reflect feedback

**Pull request merged**